### ISSUE TITLE
Ad-blocking no longer works on Opera ("Error: Internal error" while updating DNR lists)

### ISSUE THREAD

**Issue by philipp-classen** [ghostery/ghostery-extension#1554]
**TL;DR: If you are on Opera and Ad-Blocking does not work, try to uninstall and reinstall Ghostery.**

---

On Opera, a profile can enter a corrupt state where the DNR lists cannot be enabled:
```
dnr.js:49 DNR - Error while updating lists: Error: Internal error.
```

This is the API call that fails:
https://github.com/ghostery/ghostery-extension/blob/5fca91b48721da0071286f33fa6001f0ae2e102a/extension-manifest-v3/src/background/dnr.js#L42

The internal error could be reproduced by this statement as well:
```
await chrome.declarativeNetRequest.updateEnabledRulesets({  enableRulesetIds: 'ads' });
```

It appears to be a bug in the browser. The only workaround that we have found is to uninstall the extension and reinstall it. The DNR rules are static rules (part of the release). Since we failed to reproduce it yet with a fresh profile, it might be triggered by an extension update.

Opera: 109.0.5097.45 (Arch Linux)

**Comment by philipp-classen** [comment]
Could be this error: https://issues.chromium.org/issues/40285683 (fixed one month ago in Chromium Canary)

**Comment by philipp-classen** [comment]
For those who are affected: until it is fixed in the browser, there is nothing that you can do instead of uninstalling and reinstalling the extension.

On a broken profile, I got now a situation where the page `https://www.idealo.de/` is unusable (a modal dialog being over the whole page, making it non-interactive). But this problem does not occur on the other Opera profile, where I did reinstall the extension.

**Comment by philipp-classen** [comment]
Since we just a release: on my broken, updating the adblocker engine forced a reset.

Based on that knowledge, I expect that if Opera ships the bugfix (from the Chromium code base), the problem should eventually vanish, even without user action.

**Comment by philipp-classen** [comment]
Closing it now, since it is likely enough that the fix has been propagated to all browsers that are Chromium forks.

Should the error (`Error while updating lists: Error: Internal error.`) still occur, we can reopen.

**Comment by philipp-classen** [comment]
Reopening because of https://github.com/ghostery/ghostery-extension/issues/1653#issuecomment-2168337684

**Comment by philipp-classen** [comment]
Adding Chrome and Edge, but it may affect all Chromium forks, not only Opera.

Background: Opera was the first platform where we rolled out Ghostery 10 (Manifest V3). Thus, Opera used to be the only version that could be hit by the browser bug.

**Comment by smalluban** [comment]
As there is no discussion about the subject, and the issue is already fixed in current versions of the browsers, we can close the issue.
